src/plugins/data_source_management/public/components/validation/datasource_form_validation.ts

```diff
@@ -50,8 +50,14 @@ export const isTitleValid = (
     error: '',
   };
   /* Title validation */
-  if (!title?.trim?.().length) {
+  if (!title.trim().length) {
```

Comment:
Just curious whether title could be `undefined`. If yes, then calling `trim` function will throw exception. If no, then there is no concern.

Reply:
title will not be undefined, it's initialized with empty string
that's why in the function param, we didn't declare it to be optional param
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/eef417c03e7c4d34e85c70edb6c2079528ebc683/src/plugins/data_source_management/public/components/create_data_source_wizard/components/create_form/create_data_source_form.tsx#L98